code/model/editableformfields/EditableFileField.php

```diff
@@ -49,9 +49,14 @@ public function getCMSFields() {
 	}
 
 	public function getFormField() {
-		$field = FileField::create($this->Name, $this->EscapedTitle)
-			->setFieldHolderTemplate('UserFormsField_holder')
-			->setTemplate('UserFormsFileField');
+		if(isset(Config::inst()->get('EditableFileField', 'use_uploadfield')) && Config::inst()->get('EditableFileField', 'use_uploadfield')) {
```

Comment:
This isn't compatible in PHP versions that this module supports.

It's also unrelated to fixing the routing, so it seems like this should be raised as a separate pull request to this one.

Also I don't like this solution; It feels awkward where you could simply have made the classname a configurable option instead. :)